src/Symfony/Component/Translation/MessageCatalogue.php

```diff
@@ -158,9 +158,16 @@ public function replace($messages, $domain = 'messages')
     public function add($messages, $domain = 'messages')
     {
         if (!isset($this->messages[$domain])) {
-            $this->messages[$domain] = $messages;
-        } else {
-            foreach ($messages as $id => $message) {
+            $this->messages[$domain] = [];
+        }
+        $intlDomain = $domain;
+        if (false === mb_strpos($intlDomain, self::INTL_DOMAIN_SUFFIX)) {
```

Comment:
Using mbstring here is wrong (general note: don't use it blindly).

The check should be like:
```php
false === strpos($domain, self::INTL_DOMAIN_SUFFIX, -\strlen(self::INTL_DOMAIN_SUFFIX))
```

Reply:
This can lead to `Warning: strpos(): Offset not contained in string`
How about omitting the offset argument and only changing `mb_strpos` to `strpos`?

Reply:
-> `if (false === \strpos($domain, self::INTL_DOMAIN_SUFFIX)) {`

Reply:
that'd be less strict, we only want to check the end of the domain

Reply:
Then let's do this:
```php
        $suffixLength = \strlen(self::INTL_DOMAIN_SUFFIX);
        if (\strlen($domain) > $suffixLength && false !== strpos($domain, self::INTL_DOMAIN_SUFFIX, -$suffixLength)) {
            $intlDomain .= self::INTL_DOMAIN_SUFFIX;
        }
```

Reply:
Updated in 2f79462

Reply:
perfect